src/code/z_message.c

```diff
@@ -44,17 +213,106 @@
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_8014C70C.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_8014CC14.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_8014CCB4.s")
+void Message_LoadChar(GlobalContext* globalCtx, u16 codePointIndex, s32* offset, f32* arg3, s16 decodedBufPos) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    s32 temp1 = *offset;
+    f32 temp2 = *arg3;
+
+    Font_LoadChar(globalCtx, codePointIndex, temp1);
+    ((u16*)msgCtx->decodedBuffer)[decodedBufPos] = codePointIndex;
+    temp1 += 128;
+    temp2 += (16.0f * msgCtx->unk12098);
+    *offset = temp1;
+    *arg3 = temp2;
+}
+
+// Message_LoadRupees JPN ?
+void func_8014CCB4(GlobalContext* globalCtx, s16* decodedBufPos, s32* offset, f32* arg3) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    s16 t = *decodedBufPos;
+    s32 k = *offset;
+    f32 f = *arg3;
+
+    Font_LoadChar(globalCtx, 0x838BU, k);
+    k += FONT_CHAR_TEX_SIZE;
+    ((u16*)msgCtx->decodedBuffer)[t] = 0x838B;
+    t += 1;
+    Font_LoadChar(globalCtx, 0x8373U, k);
+    k += FONT_CHAR_TEX_SIZE;
+    ((u16*)msgCtx->decodedBuffer)[t] = 0x8373;
+    t += 1;
+    Font_LoadChar(globalCtx, 0x815CU, k);
+    k += FONT_CHAR_TEX_SIZE;
+    ((u16*)msgCtx->decodedBuffer)[t] = 0x815C;
```

Comment:
```suggestion
    Font_LoadChar(globalCtx, 0x838B, k);
    k += FONT_CHAR_TEX_SIZE;
    ((u16*)msgCtx->decodedBuffer)[t] = 0x838B;
    t += 1;
    Font_LoadChar(globalCtx, 0x8373, k);
    k += FONT_CHAR_TEX_SIZE;
    ((u16*)msgCtx->decodedBuffer)[t] = 0x8373;
    t += 1;
    Font_LoadChar(globalCtx, 0x815C, k);
    k += FONT_CHAR_TEX_SIZE;
    ((u16*)msgCtx->decodedBuffer)[t] = 0x815C;
```
Is the `U` needed? If not they should be removed.

Reply:
not needed, removed it and replaced some of those constant to their ascii equivalent.
in `z_message.c` those write `ルピ―` (rupees in japanese) using JISX0213 standard, couldn't write those directly in the code because the compiler use EUC_JP encoding which uses other codepoints for those characters. Added that info as a comment because I don't know enough about characters encoding to really know how to handle encoding translation/transform